### DISCUSSION TITLE
Adding additional rulesets to so-octopus/Playbooks does not result in them being imported (2.4.20)

### DISCUSSION THREAD

**Discussion by thedeadliestcatch** [Security-Onion-Solutions/securityonion#11593] [2.4] [answered]
I strongly disagree with the conversion to discussion:

- The issue manifests following the documentation step by step (https://docs.securityonion.net/en/2.4/playbook.html).
- The newly enabled rulesets appear in the /opt/so/saltstack/local/pillar/global.sls file after synchronizing the grid.
- _so-playbook-ruleupdate_ **ignores** the above and displays very clearly that only the 'windows' category is enabled.
-  Successive runs do not change this fact.
- A reboot does not fix the issue.
- Successive salt highstate calls do not fix the issue.

The VM is provisioned with resources exceeding the requirements for the manager node and there are no indicators of broken components, connectivity issues, etc. Everything seemingly completes and runs as it should, except that the update for so-octopus rulesets alters  /opt/so/saltstack/local/pillar/global.sls correctly but results in a no-op: the settings are not propagated.

As I understand it, from reviewing everything mentioning non-windows playbooks, this seems like a heavily untested area and it is quite likely that the support there is broken in 2.4.20. I recall a particular issue/thread where this is explicitly mentioned ("no support yet").

If this is the case, the documentation should be amended accordingly so that the expectations are set ("do not expect additional rulesets to work, be integrated/syncd, ...".

In that case, it might be helpful to provide actual troubleshooting steps and manual configuration notes, as you are leveraging docker containers and this might very well be an issue related to overlayfs and how you handle the volumes.

### Discussed in https://github.com/Security-Onion-Solutions/securityonion/discussions/11590

<div type='discussions-op-text'>

<sup>Originally posted by **thedeadliestcatch** October 22, 2023</sup>
I have properly followed the instructions at https://docs.securityonion.net/en/2.4/playbook.html:

> Go to [Administration](https://docs.securityonion.net/en/2.4/administration.html#administration) –> Configuration.
>     At the top of the page, click the Options menu and enable the Show all configurable settings, including advanced settings. option.
>     On the left side, navigate to soctopus –> playbook –> rulesets.
>     On the right side, add one or more of the following: application,category,cloud,compliance,linux,macos,network,web,windows. These are based on the top level directories from the Sigma community repository rule’s folder.
>     At the top of the page, click the SYNCHRONIZE GRID button under the Options menu.

sooctopus is enabled and I have synchronized the grid manually.

The rulesets added included "linux" and others.

Unfortunately, none other than the default IDH and windows rulesets are to be found in Playbooks. The expected behavior is to see the new rules imported in Draft status.

</div>

**Comment by thedeadliestcatch** [comment]
To further illustrate the **bug**:


```
-= Update Summary =-

Sigma Community Repo:
 Already up to date.

Updated Plays: 2
Updates Available: 0
New Plays: 81
No Updates Needed: 2083

Enabled Rulesets:
['windows']

-= Completed: 2023-10-22 ...
```

```
# cat /opt/so/saltstack/local/pillar/soctopus/soc_soctopus.sls 
soctopus:
    enabled: true
    playbook:
        rulesets:
            - windows
            - web
            - linux
            - application
            - category
            - network

```

We can safely assume that updating the rulesets through the UI and starting a grid sync does successfully update the sls. We can also assume that this is not propagated further down and that all successive runs of so-playbook-ruleupdate continue ignoring the updated rulesets setting.

**Comment by TOoSmOotH** [comment]
We troubleshoot everything as a discussion then move it to an issue once it is confirmed by one of the devs. We then create an issue based off the discussion for the fix. We appreciate you finding a potential bug and by us moving it to a discussion was in no way meant to say it isn't. This is just the process.

**Comment by TOoSmOotH** [comment, marked as answer]
https://github.com/Security-Onion-Solutions/securityonion/pull/11594

In the mean time you should be able to do the following on the manager:

sudo docker stop so-soctopus
sudo docker rm so-soctopus
sudo salt-call state.apply soctopus
